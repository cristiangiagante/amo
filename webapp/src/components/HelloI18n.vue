<template>
  <p>{{ $t('hello') }}</p>
</template>

<script>
export default {
  name: 'HelloI18n'
}
</script>

<i18n>
{
  "en": {
    "hello": "Hello i18n in SFC!"
  },
  "es": {
    "hello": "Hola en i18n en un SFC!"
  }
}
</i18n>
