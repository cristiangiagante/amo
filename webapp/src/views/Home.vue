<template>
<b-container class="bv-example-row">
    <b-row>
        <b-col>1 of 2</b-col>
        <b-col>2 of 2</b-col>
    </b-row>
    <b-row>
        <b-col>1 of 3</b-col>
        <b-col>2 of 3</b-col>
        <b-col>3 of 3</b-col>
    </b-row>
</b-container>
</template>

<script>
// @ is an alias to /src
import Turns from '@/components/Turns.vue'

export default {
  name: 'home',
  components: {
    Turns
  }
}
</script>
