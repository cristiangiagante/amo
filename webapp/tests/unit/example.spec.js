import { shallowMount, createLocalVue } from '@vue/test-utils'
import Home from '@/views/Home.vue'
import HelloI18n from '@/components/HelloI18n.vue'
import BootstrapVue from 'bootstrap-vue'
import VueI18n from 'vue-i18n'

describe('Home.vue', () => {
  it('renders props.msg when passed', () => {
    let localVue = createLocalVue()
    localVue.use(BootstrapVue)
    const wrapper = shallowMount(Home, {
      localVue
    })
    expect(wrapper.text()).toMatch('1 of 3')
  })

  it('renders language spanish in i18n', () => {
    let localVue = createLocalVue()
    localVue.use(VueI18n)
    const messages = {
      en: {
        hello: 'Enter a name'
      },
      es: {
        hello: 'hola'
      }
    }
    const i18n = new VueI18n({
      locale: 'es',
      fallbackLocale: 'es',
      messages
    })
    const wrapper = shallowMount(HelloI18n, {
      i18n,
      localVue
    })
    expect(wrapper).toMatchSnapshot()
  })
})
