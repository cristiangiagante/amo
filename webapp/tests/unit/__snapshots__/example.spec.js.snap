// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders language spanish in i18n 1`] = `<p>hola</p>`;
